src/SharedPipelineComponents/Datalock/SFA.DAS.CollectionEarnings.DataLock/Services/DatalockValidationService.cs

```diff
@@ -0,0 +1,295 @@
+        public List<DatalockValidationErrorByPeriod> ValidationErrorsByPeriod { get; } = new List<DatalockValidationErrorByPeriod>();
+        public List<PriceEpisodePeriodMatchEntity> PriceEpisodePeriodMatches { get; } = new List<PriceEpisodePeriodMatchEntity>();
+        public List<PriceEpisodeMatchEntity> PriceEpisodeMatches { get; } = new List<PriceEpisodeMatchEntity>();
+        public List<DatalockOutputEntity> DatalockOutputEntities { get; } = new List<DatalockOutputEntity>();
+        
+        public void AddDistinctRecords(RawEarning earning, List<string> errors, TransactionTypesFlag paymentType, CommitmentEntity commitment)
+        {
+            var payable = false;
+            if (errors.Count > 0)
+            {
+                foreach (var error in errors)
+                {
+                    if (ValidationErrors.FirstOrDefault(x =>
+                            x.LearnRefNumber == earning.LearnRefNumber &&
+                            x.AimSeqNumber == earning.AimSeqNumber &&
+                            x.PriceEpisodeIdentifier == earning.PriceEpisodeIdentifier &&
+                            x.RuleId == error &&
+                            x.Ukprn == earning.Ukprn) == null &&
+                        PriceEpisodePeriodMatches.FirstOrDefault(x => 
+                            x.AimSeqNumber == earning.AimSeqNumber &&
+                            x.LearnRefNumber == earning.LearnRefNumber &&
+                            x.PriceEpisodeIdentifier == earning.PriceEpisodeIdentifier &&
+                            x.Ukprn == earning.Ukprn && 
+                            x.CommitmentId == commitment.CommitmentId &&
+                            x.Period == earning.Period &&
+                            x.TransactionTypesFlag == paymentType &&
+                            x.Payable &&
+                            x.VersionId == commitment.VersionId) == null)
+                    {
+                        ValidationErrors.Add(new DatalockValidationError
+                        {
+                            LearnRefNumber = earning.LearnRefNumber,
+                            AimSeqNumber = earning.AimSeqNumber,
+                            PriceEpisodeIdentifier = earning.PriceEpisodeIdentifier,
+                            RuleId = error,
+                            Ukprn = earning.Ukprn,
+                        });
+                    }
+
+                    if (ValidationErrorsByPeriod.FirstOrDefault(x =>
+                            x.LearnRefNumber == earning.LearnRefNumber &&
+                            x.AimSeqNumber == earning.AimSeqNumber &&
+                            x.PriceEpisodeIdentifier == earning.PriceEpisodeIdentifier &&
+                            x.RuleId == error &&
+                            x.Period == earning.Period &&
+                            x.Ukprn == earning.Ukprn) == null)
+                    {
```

Comment:
this looks similar or the same as the statement above. So a private or extension method might be clearer